### ISSUE TITLE
ical export uid assignment causes issues if you've imported a hyperschedule-generated ical file before

### ISSUE THREAD

**Issue by apomyxis** [hyperschedule/hyperschedule#114]
pretty sure it's related to [this thing about UIDs in ical files](https://en.wikipedia.org/wiki/ICalendar#Distributing_updates)

basically when i imported the ical file a bunch of my classes from last semester disappeared from my several months ago calendar because google calendar tried to "update" the event with the matching UID. in addition, some of the "newly imported" events appeared with identical name, description, room, etc to my classes from last semester (presumably attempted "updates" from those existing events?) rather than the correct courses for this semester.

the solution is probably just to generate an actually unique ID for each course(?)

**Comment by Maxwellfire** [comment]
Was about to create this exact issue reporting the same thing. I agree that a unique UID for each course should fix the problem

**Comment by Maxwellfire** [comment]
Traced this to line 179 in [here](https://github.com/nwcell/ics.js/blob/master/ics.js) , which I believe is the ICS writing library we are using. 

`'UID:' + calendarEvents.length + "@" + uidDomain,`

To fix this, changes would have to be pulled into that code base. I don't know why they thought creating not unique identifiers would be a good idea

**Comment by kwshi** [comment]
Indeed--there are many mediocre/bad things about the iCal library we use, and we very well may think about using a different one/rolling our own at some point soon.

**Comment by raxod502** [comment]
> To fix this, changes would have to be pulled into that code base.

I don't think so. We should just need to call `ics(functionToMakeActualUUID())` instead of `ics()`, right?

**Comment by Maxwellfire** [comment]
Oh I guess we could use the UID domain to accomplish that. On that point, something like `hyperscheduler_sp_2019` might be good. So that if you re-import classes it updates them instead of duplicating them, but doesn't carry over from semester to semester

**Comment by raxod502** [comment]
I don't really like that idea. Note that the UID generation uses the _index_ of the course (which makes no sense to me). So if you add, remove, or rearrange courses and then re-export the schedule, which events will be updated versus created will be totally bizarre.
